src/blockrelay/graphene.cpp

```diff
@@ -755,10 +775,15 @@ static bool ReconstructBlock(CNode *pfrom,
     // Add the header size to the current size being tracked
     thinrelay.AddBlockBytes(::GetSerializeSize(pblock->GetBlockHeader(), SER_NETWORK, PROTOCOL_VERSION), pblock);
 
+    // If we have incomplete infomation about this block, resize the block transaction count to accomodate new data
+    if (pblock->vtx.size() < grapheneBlock->vTxHashes256.size())
```

Comment:
This also seems like a red flag to me. I would be more inclined to throw an error if `pblock->vtx.size() != grapheneBlock->vTxHashes256.size()`. But do you think that there is a legitimate scenario where this can happen?

Reply:
OK  , I see, if blockCheapHashes returns incorrectly from the set reconciliation then this scenario could happen.  The reconstruction will likely fail with this new resizing but at least we won't segfault.  Changing this to != doesn't matter , but either way is fine with me.